eclair-core/src/main/scala/fr/acinq/eclair/router/Graph.scala

```diff
@@ -331,7 +331,7 @@ object Graph {
         val totalHopsCost = prev.virtualFees + hopCost
         // If the edge was added by the invoice, it is assumed that it can route the payment.
         // If we know the balance of the channel, then we will check separately that it can relay the payment.
-        val successProbability = if (edge.update.chainHash == ByteVector32.Zeroes || edge.balance_opt.nonEmpty) 1.0 else 1.0 - prev.amount.toLong.toDouble / edge.capacity.toMilliSatoshi.toLong.toDouble
+        val successProbability = if (edge.source.isInstanceOf[ChannelSource.Hint] || edge.balance_opt.nonEmpty) 1.0 else 1.0 - prev.amount.toLong.toDouble / edge.capacity.toMilliSatoshi.toLong.toDouble
```

Comment:
For hints, you already set balance_opt so you don't need this anymore.
```suggestion
        val successProbability = if (edge.balance_opt.nonEmpty) 1.0 else 1.0 - prev.amount.toLong.toDouble / edge.capacity.toMilliSatoshi.toLong.toDouble
```

Reply:
Why weren't we able to do so before this PR?

Reply:
You're right, it could have already been done before.
Before calling `addEdgeWeight`, we've already checked that the amount is less than `edge.balance_opt` if it is defined. So if `edge.balance_opt` is defined here, it means that we know the balance and it is high enough. `edge.source.isInstanceOf[ChannelSource.Hint]` was a bit of a hack, the thing we care about is if the `balance_opt` is defined an larger than the amount.